packages/beacon-node/src/api/impl/debug/index.ts

```diff
@@ -85,5 +86,28 @@ export function getDebugApi({
         },
       };
     },
+    async getHistoricalSummaries({stateId}, _context) {
+      const {state} = await getStateResponseWithRegen(chain, stateId);
+      let slot: number;
+      if (state instanceof Uint8Array) {
+        slot = getStateSlotFromBytes(state);
+      } else {
+        slot = state.slot;
+      }
+      if (config.getForkSeq(slot) < ForkSeq.capella) {
+        throw new Error("Historical summaries are not supported before Capella");
+      }
+      const fork = config.getForkName(slot) as Exclude<ForkName, "phase0" | "altair" | "bellatrix">;
+      const stateView = state instanceof Uint8Array ? ssz[fork].BeaconState.deserializeToViewDU(state) : state;
```

Comment:
This needs to follow similar pattern as here
https://github.com/ChainSafe/lodestar/blob/64eb0153dd97b7525cb61625b102a6a521869c05/packages/beacon-node/src/api/impl/validator/index.ts#L1021-L1024

otherwise we it will be pretty expensive memory wise

Reply:
Fixed